pkg/scheduler/framework/plugins/nodevolumelimits/csi.go

```diff
@@ -66,6 +67,17 @@ func (pl *CSILimits) Name() string {
 	return CSIName
 }
 
+// EventsToRegister returns the possible events that may make a Pod
+// failed by this plugin schedulable.
+func (pl *CSILimits) EventsToRegister() []framework.ClusterEvent {
+	return []framework.ClusterEvent{
+		{Resource: framework.CSINode, ActionType: framework.Add | framework.Update},
+		{Resource: framework.PersistentVolume, ActionType: framework.Add | framework.Update},
+		{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update},
+		{Resource: framework.StorageClass, ActionType: framework.Add},
```

Comment:
Similar to my previous comment:

- PodDelete should be counted, as `currentVolumeCount` may decrease upon deletion of existing pods.
- PodUpdate _may_ be counted if pod's volume is mutable (need sig-storage experts to confirm)
- PV / PVC / SC events don't seem related

Reply:
Pod volumes are not mutable.

I think we don't need the PV/PVC/SC events. We will not count a volume if any of these items are potentially missing, so an update fixing that condition would only make it count towards the limit, which doesn't help with making the pod schedulable.

Same for the non-csi predicate.

Reply:
Thanks @msau42!

Could you help also confirm https://github.com/kubernetes/kubernetes/pull/100026#discussion_r598443351? (my guess is no)

Reply:
@minbaev could you address the comments?

- PV / PVC / SC events are not needed
- PodDelete should be counted, but not for PodUpdate

@msau42 could you confirm that if Node Add event is needed? I suppose the answer is no, as this plugin seems to rely on CSINode events only.

PS: csi.go#getVolumeLimits() stated:

```
// TODO: stop getting values from Node object in v1.18
```

We can get it cleaned up in another PR.

Reply:
For csi.go, we don't need Node events (only CSINode)

Reply:
@msau42 thanks for the confirmation.